---
title: "04_RandomForest"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Set up

### Load libraries
```{r}
library(mctoolsr)
library(plyr) 
library(tidyverse) 
library(vegan)
library(plotly)
library(readr)
library(caret)
library(ranger)
library(stringr)
library(Boruta)

set.seed(10)
```

### Set up input and output directories
```{r Set up input and output directories, include = FALSE}
project.fp <- "/Users/coin8046/Desktop/FiererLabNotebook_copy/mustard_microbiome/mustard_ITS/R.analyses"
clean_data.fp <- file.path(project.fp, "01_clean_data")
    outputs_01_clean.fp <- file.path(clean_data.fp, "outputs")

explr_02.fp <- file.path(project.fp, "02_exploration")
    outputs_02.fp <- file.path(explr_02.fp, "outputs")

randfor_04.fp <- file.path(project.fp, "04_RandomForest")
  outputs_04.fp <- file.path(randfor_04.fp, "outputs")
  figures_04.fp <- file.path(randfor_04.fp, "figures")

if (!dir.exists(randfor_04.fp)) {dir.create(randfor_04.fp, recursive = TRUE)}    
if (!dir.exists(outputs_04.fp)) {dir.create(outputs_04.fp, recursive = TRUE)}
if (!dir.exists(figures_04.fp)) {dir.create(figures_04.fp, recursive = TRUE)}
```

## Data set up 

### Read in ITS data
```{r}
# rhizosphere data
## rarefied, filtered data: chem samples only

input_rhizo.chem <- readRDS(paste0(outputs_02.fp, "/input_rhizo.chem.RDS"))
input_rhzC.M <- readRDS(paste0(outputs_02.fp, "/input_rhzC.M.RDS"))
input_rhzC.S <- readRDS(paste0(outputs_02.fp, "/input_rhzC.S.RDS"))

# root data
#input_root.chem <- readRDS(paste0(outputs_02.fp, "/input_root.chem.RDS"))
#input_rtC.M <- readRDS(paste0(outputs_02.fp, "/input_rtM.com.RDS"))
#input_rtC.S <- readRDS(paste0(outputs_02.fp, "/input_rtS.com.RDS"))
```

### Read in 16S data for combo modeling
```{r}
project16S.fp <- "/Users/coin8046/Desktop/FiererLabNotebook_copy/mustard_microbiome/mustard_16S/r_analyses"
explr_02_16S.fp <- file.path(project16S.fp, "02_exploration")
    outputs_02_16S.fp <- file.path(explr_02_16S.fp, "outputs")

# rhizosphere 16S data    
input16S_rhizo.c <- readRDS(paste0(outputs_02_16S.fp, "/input_rhizo.chem10.RDS"))
input16S_rhizo.cM <- readRDS(paste0(outputs_02_16S.fp, "/input_rhizo.cM10.RDS"))
input16S_rhizo.cS <- readRDS(paste0(outputs_02_16S.fp, "/input_rhizo.cS10.RDS"))

# root 16S data  
# input16S_root.c <- readRDS(paste0(outputs_02_16S.fp, "/input_root.chem.RDS"))
# input16S_root.cM <- readRDS(paste0(outputs_02_16S.fp, "/input_root.cM.RDS"))
# input16S_root.cS <- readRDS(paste0(outputs_02_16S.fp, "/input_root.cS.RDS"))
```

### Read in sample metadata
```{r}
metadata.r <- readRDS("/Users/coin8046/Desktop/FiererLabNotebook_copy/mustard_microbiome/mustard_chem/01_outputs/plant.chem_data.RDS")

metadata <- metadata.r[-c(70), ]
sampleIDs <- metadata$Pot.ID

```

## Prep dataframes for Random Forest
### ITS
```{r}
# use input_rhzC.M

ASV_rhizo.M <- summarize_taxonomy(input_rhzC.M, level = 7, report_higher_tax =  FALSE) %>% 
  t() %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "sample_id") %>% 
  inner_join(input_rhzC.M$map_loaded, by = "sample_id")



#prepare final data frame, keeping only taxa and response variable of interest "Allyl..µmol.g."
ASV_rhizo.M.rf <- ASV_rhizo.M %>% 
  column_to_rownames(var = "sample_id") %>% 
  # dplyr::select(c("ASV_145","ASV_16","ASV_2","ASV_25","ASV_4","ASV_44","Allyl..µmol.g."))
  dplyr::select(-c("Primer_Plate", "Primer_Well", "sample_type_1", "sample_type_2", "ReadCount.raw", "condition","soil_type.x","sterility.x", "SampleID_3", "flower.buds.appear", "full.flower.appear", "seed.pods.appear", "no.flowers..pods.drying", "Spider.mites", "total.seed.count", "Last.Seeds.Collected", "Root.Rhizo.Harvest","total_seed_mass","sample.name","height_9","height_15","Updated.Mass..g.","X3.Butenyl..µmol.g.","X3MT..µmol.g.","Indoles..µmol.g.","treatment","soil_type.y","sterility.y"))

```

### 16S & combo
```{r}
## Use relative abundance 16S dataframe
ASV16S_rhizo.M.r <- summarize_taxonomy(input16S_rhizo.cM, level = 7, report_higher_tax =  FALSE) %>% 
  t() %>% as.data.frame()

## reformat names to prevent conflicts with ITS names
cols16S.r <- names(ASV16S_rhizo.M.r)
cols16S <- paste0(cols16S.r,"_16S")

oldnames = c(cols16S.r)
newnames = c(cols16S)

ASV16S_rhizo.M <- ASV16S_rhizo.M.r %>% rename_at(vars(all_of(oldnames)), ~ newnames) %>% 
  rownames_to_column(var = "sample_id")


## Combine 16S and ITS dataframes 
combo_rhizoM.rf <- ASV_rhizo.M.rf %>% 
  rownames_to_column(var = "sample_id") %>% 
  inner_join(ASV16S_rhizo.M, by = "sample_id")


######################
######################
# this is the final table to input into final RF model # 
##### filter to keep specific taxa only (as identified by Boruta, below)
combo_rhizoM.rf.filt2 <- combo_rhizoM.rf %>% 
  column_to_rownames(var = "sample_id") %>% 
  dplyr::select(c("ASV_2", "ASV_25", "ASV_65", 
                    "ASV_11_16S",  "ASV_187_16S","ASV_190_16S", 
                    "ASV_1036_16S", "ASV_1205_16S","ASV_22_16S",
                    "ASV_217_16S",  "ASV_369_16S","ASV_394_16S",
                    "ASV_45_16S", "ASV_524_16S", "ASV_692_16S", 
                   "ASV_697_16S","ASV_76_16S","ASV_871_16S",
                   "Allyl..µmol.g."))


```

### Make dataframe of sterile inocula samples to test with model
```{r}
### STERILE relative abundance 16S
ASV16S_rhizo.S.r <- summarize_taxonomy(input16S_rhizo.cS, level = 7, report_higher_tax =  FALSE) %>% 
  t() %>% as.data.frame()

Scols16S.r <- names(ASV16S_rhizo.S.r)
Scols16S <- paste0(Scols16S.r,"_16S")

Soldnames = c(Scols16S.r)
Snewnames = c(Scols16S)

ASV16S_rhizo.S <- ASV16S_rhizo.S.r %>% rename_at(vars(all_of(Soldnames)), ~ Snewnames) %>% 
  rownames_to_column(var = "sample_id")

### STERILE ITS
ASV_rhizo.S <- summarize_taxonomy(input_rhzC.S, level = 7, report_higher_tax =  FALSE) %>% 
  t() %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "sample_id") %>% 
  inner_join(input_rhzC.S$map_loaded, by = "sample_id")

ASV_rhizo.S.rf <- ASV_rhizo.S %>% 
  column_to_rownames(var = "sample_id") %>% 
  dplyr::select(-c("Primer_Plate", "Primer_Well", "sample_type_1", "sample_type_2", "ReadCount.raw", "condition","soil_type.x","sterility.x", "SampleID_3", "flower.buds.appear", "full.flower.appear", "seed.pods.appear", "no.flowers..pods.drying", "Spider.mites", "total.seed.count", "Last.Seeds.Collected", "Root.Rhizo.Harvest","total_seed_mass","sample.name","height_9","height_15","Updated.Mass..g.","X3.Butenyl..µmol.g.","X3MT..µmol.g.","Indoles..µmol.g.","treatment","soil_type.y","sterility.y"))

#### combination STERILE data
combo_rhizoS.rf <- ASV_rhizo.S.rf %>% 
  rownames_to_column(var = "sample_id") %>% 
  inner_join(ASV16S_rhizo.S, by = "sample_id")
```

### Save dfs for other modeling
```{r}
## reorder columns first
combo_rhizoM.final <- combo_rhizoM.rf %>% 
  select('sample_id', 'Allyl..µmol.g.', everything()) %>% 
  column_to_rownames(var = "sample_id")

combo_rhizoS.final <- combo_rhizoS.rf %>% 
  select('sample_id', 'Allyl..µmol.g.', everything()) %>% 
  column_to_rownames(var = "sample_id")

  
#write.csv(combo_rhizoM.final,  file = paste0(outputs_04.fp, "/ASV_rhizo.M.COMB10.csv"))
#write.csv(combo_rhizoS.final,  file = paste0(outputs_04.fp, "/ASV_rhizo.S.COMB10.csv"))
#write.csv(combo_rhizoM.rf.filt2,  file = paste0(outputs_04.fp, "/ASV_rhizo.M.10filt_20Mar23.csv"))

#combo_rhizoM.final <- read_csv(paste0(outputs_04.fp, "/ASV_rhizo.M.COMB10.csv"))
#combo_rhizoS.final <- read_csv(paste0(outputs_04.fp, "/ASV_rhizo.S.COMB10.csv"))

#combo_rhizoM.rf.filt2 <- read_csv(paste0(outputs_04.fp, "/ASV_rhizo.M.10filt_20Mar23.csv"))
```


## Boruta feature selection
```{r}
#https://academic.oup.com/bib/article/20/2/492/4554516
#https://www.machinelearningplus.com/machine-learning/feature-selection/

# Perform Boruta search (ITS only)
boruta_output.rz.M <- Boruta(Allyl..µmol.g. ~., 
                        data=na.omit(ASV_rhizo.M.rf), doTrace=0) 
boruta_output.rz.M
boruta_signif.rz.M <- getSelectedAttributes(boruta_output.rz.M, withTentative = TRUE)
#"ASV_145" "ASV_16"  "ASV_2"   "ASV_25"  "ASV_4"   "ASV_44" 

# Perform Boruta search (ITS & 16S)
boruta_output.comb.rz.M <- Boruta(Allyl..µmol.g. ~., 
                        data=na.omit(combo_rhizoM.rf), doTrace=0) 
boruta_output.comb.rz.M
boruta_sig.comb.rz.M <- getSelectedAttributes(boruta_output.comb.rz.M, withTentative = TRUE)

```


## Read in final dataframes
### Split into training and test sets
```{r}
## read in filtered 16S and ITS dataframe
combo_rhizoM.rf.filt2 <- read_csv(paste0(outputs_04.fp, "/ASV_rhizo.M.10filt_20Mar23.csv"))
colnames(combo_rhizoM.rf.filt2)[1] = "sample_id"

## filter to remove unwanted metadata, keeping only response variable of interest
combo_rhizoM.rf.filt2t <- combo_rhizoM.rf.filt2 %>% 
  mutate(sample_id2 = gsub('(.*)_\\w+', '\\1', sample_id)) %>% 
  inner_join(metadata, by = c("sample_id2" = "Pot.ID")) %>% 
  # keep  #  "total_seed_mass", "height_9"  
  select(-c("sample_id2", "flower.buds.appear", "full.flower.appear", "seed.pods.appear",
            "no.flowers..pods.drying",  "Spider.mites", "total.seed.count" , "Last.Seeds.Collected", "Root.Rhizo.Harvest", "sample.name",  "height_15", "Updated.Mass..g.", "X3.Butenyl..µmol.g.", "X3MT..µmol.g.", "Indoles..µmol.g.", "treatment", "soil_type", "sterility", "Allyl..µmol.g..y")) %>% 
  column_to_rownames(var = "sample_id")

## this dataframe is for testing other plant attributes as predictive in model
#combo_rhizoM.rf.filt2t <- combo_rhizoM.rf.filt2t[, c(1:18, 21,20,19)]


              
### Separate training and testing data
set.seed(10) # iterate over 10 different seeds for cross validation

random_ordered.comb <- combo_rhizoM.rf.filt2[sample(nrow(combo_rhizoM.rf.filt2)),]
number_training_samples.comb <- ceiling(nrow(combo_rhizoM.rf.filt2) * 0.6)
train.comb <- random_ordered.comb[1:number_training_samples.comb,]
test.comb <- random_ordered.comb[(number_training_samples.comb + 1):nrow(random_ordered.comb),]
```

## Random forest training
### ITS & 16S
```{r}

### Create random forest models
## tune grid
tunegrid.comb <- data.frame(mtry = c(2, 3, 4), #2,3,4
                             min.node.size = c(2, 4, 10),#2,4,10
                             splitrule = c('variance'))

### hyperparamters for train.1
# control <- trainControl(method='repeatedcv', 
#                         number=10, 
#                         repeats=3) 
#                         #seeds =)
# 
# mtry.comb <- sqrt(ncol(train.comb))


## train model
train1.comb <- train(Allyl..µmol.g..x ~.,
                data=train.comb,
                method="ranger",
                metric = "Rsquared",
                num.trees = 1000,
                tuneGrid = tunegrid.comb, 
                #trControl =control, 
                seed = 10) 
print(train1.comb)

## predict
pred2 <- predict(train1.comb, newdata = test.comb[1:ncol(test.comb)-1]) %>% as.data.frame() %>% rownames_to_column(var = "sample_num")
cor.test(pred2$., test.comb$Allyl..µmol.g.)



saveRDS(train1.comb, file = paste0(outputs_04.fp, "/RFmodel_3.20.23.RDS"))


# another way, and other hyperparameters
ranger1.comb <- ranger(Allyl..µmol.g..x ~.,
                  data=train.comb,
                  num.trees = 1000,
                  splitrule = "variance",
                  write.forest = TRUE, 
                  classification = FALSE) 
print(ranger1.comb)


```


### Test predictions with test data
```{r}
#predict based on original model
pred2 <- predict(train1.comb, newdata = test.comb[1:ncol(test.comb)-1]) %>% as.data.frame() %>% rownames_to_column(var = "sample_num")
cor.test(pred2$., test.comb$Allyl..µmol.g.)

# predict based on alertanative model
pred <- predict(ranger1.comb, data = test.comb[1:ncol(test.comb)-1])

pred.1 <- pred$predictions %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "test_num")

cor.test(pred$predictions, test.comb$Allyl..µmol.g.)

# combine predictions from both model sources
check1 <- test.comb %>% 
  rownames_to_column(var = "sample_num") %>% 
  rownames_to_column(var = "test_num") %>% 
  select(c("Allyl..µmol.g.", "sample_num", "test_num")) %>% 
  inner_join(pred.1, by= "test_num") %>% 
  inner_join(pred2, by = c("test_num" = "sample_num"))

cor.test(check1$..x, check1$Allyl..µmol.g.)
cor.test(check1$..y, check1$Allyl..µmol.g.)

# plot predictions from original model against actual values
predcor <- ggplot(check1, aes(x=..y, y=Allyl..µmol.g.))+ #Allyl..µmol.g.
  geom_point()+
  geom_smooth(method = "lm")+
  #geom_abline(slope = 1, intercept = 0, color = 'red')+
  ylab("True proportion of Allyl in seeds")+
  xlab("Predicted proportion of Allyl in seeds")+
  #labs(title = "Predicted by slurry bacterial and fungal ASVs")+
  theme_bw(base_size = 18)
  #ggtitle("Predictive accuracy of Random Forest model")

predcor

ggsave(predcor, filename = paste0(figures_04.fp, "/predcorRFcomb3.20.png"))
```

### Test model with sterile data
```{r}
#combo_rhizoS.final

predS <- predict(train1.comb, newdata = combo_rhizoS.final[2:ncol(combo_rhizoS.final)]) %>% as.data.frame() %>% rownames_to_column(var = "sample_num")
cor.test(predS$., combo_rhizoS.final$Allyl..µmol.g.)

```

## Correlation loop
### Loop over major predictors in model to look at individual relationships
```{r}
# combo_rhizoM.rf.filt2

### this is to get the final table structure in order
## this is a list of all the ASVs in my dataframe
rf.asv <- combo_rhizoM.rf.filt2[, 1:18] %>% names()

## length of the asv list (so the loop knows how many times to loop)
iterations_rfasv = length(rf.asv)
variables = 2

## make an empty matrix to store results
rfasv_output <- matrix(ncol=variables, nrow=iterations_rfasv)

## loop
for (i in 1:iterations_rfasv){
tmp_cor <- cor.test(x = combo_rhizoM.rf.filt2[,i], y = combo_rhizoM.rf.filt2[,19], method = "pearson")
tmp_row <- c(tmp_cor$p.value, tmp_cor$estimate)
rfasv_output[i,] <- tmp_row 
}
rfasv_output <- as.data.frame(rfasv_output)
names(rfasv_output) <- c("pval", "R")
rownames(rfasv_output) <- rf.asv

## reformat output, adjust for false detection rate
rfasv_output_fdr <- rfasv_output %>% 
  rownames_to_column(var="asv") %>% 
  mutate(pval.adj = p.adjust(pval, method = "fdr")) %>% 
  mutate(sig = ifelse(pval.adj < 0.001, yes = "***", no = 
                        ifelse(pval.adj < 0.01, yes = "**", no = 
                                 ifelse(pval.adj < 0.05, yes = "*", no = "ns"))))

# reorder columns
rfasv_output_fdr <- rfasv_output_fdr[,c(1,3,2,4,5)]

# reorder rows
rfasv_output_final <- rfasv_output_fdr %>% 
  arrange(pval.adj)

write.csv(rfasv_output_final, file = paste0(outputs_04.fp, "/rfASV_cortab_3.20.23.csv"))
```

#### Check taxonomy of major predictive taxa
```{r}
## check taxonomy
model_ASVs <- c("ASV_2", "ASV_25", "ASV_65", 
                    "ASV_11_16S",  "ASV_187_16S",
                    "ASV_190_16S", "ASV_1036_16S", "ASV_1205_16S",
                    "ASV_22_16S","ASV_217_16S",  "ASV_369_16S","ASV_394_16S",
                    "ASV_45_16S", "ASV_524_16S", "ASV_692_16S", 
                   "ASV_697_16S","ASV_76_16S","ASV_871_16S")

ITS.M_ASVs <- c("ASV_2", "ASV_25", "ASV_65")
l16S.M_ASVs <- c("ASV_11",  "ASV_187",
                    "ASV_190", "ASV_1036", "ASV_1205",
                    "ASV_22","ASV_217",  "ASV_369","ASV_394",
                    "ASV_45", "ASV_524", "ASV_692", 
                   "ASV_697","ASV_76","ASV_871")


# get full taxonomy information from taxa tables
input_rhz.RF.M.its <- filter_taxa_from_input(input = input_rhzC.M, taxa_IDs_to_keep = ITS.M_ASVs)
View(input_rhz.RF.M.its$taxonomy_loaded)

input_rhz.RF.M.16s <- filter_taxa_from_input(input = input16S_rhizo.cM, taxa_IDs_to_keep = l16S.M_ASVs)
View(input_rhz.RF.M.16s$taxonomy_loaded)
```

### Plot correlations
```{r}
combo_rhizoM.rf_long <- combo_rhizoM.rf.filt2 %>% 
    pivot_longer(! Allyl..µmol.g., 
               names_to = "ASV", values_to = "abun")

## reorganize based on 16S / ITS
## color based on... taxonomy?
rhizoM_cor <- ggplot(data = combo_rhizoM.rf_long %>% 
         filter(ASV %in% c("ASV_2", "ASV_25", "ASV_65", 
                    "ASV_11_16S",  "ASV_187_16S",
                    "ASV_190_16S", "ASV_1036_16S", "ASV_1205_16S",
                    "ASV_22_16S","ASV_217_16S",  "ASV_369_16S","ASV_394_16S",
                    "ASV_45_16S", "ASV_524_16S", "ASV_692_16S", 
                   "ASV_697_16S","ASV_76_16S","ASV_871_16S",
                   "Allyl..µmol.g.")),
                aes(x = `Allyl..µmol.g.`, y = abun))+
  geom_point()+
  facet_wrap(~factor(ASV, levels=c("ASV_2", "ASV_25", "ASV_65", 
                    "ASV_11_16S",  "ASV_187_16S",
                    "ASV_190_16S", "ASV_1036_16S", "ASV_1205_16S",
                    "ASV_22_16S","ASV_217_16S",  "ASV_369_16S","ASV_394_16S",
                    "ASV_45_16S", "ASV_524_16S", "ASV_692_16S", 
                   "ASV_697_16S","ASV_76_16S","ASV_871_16S",
                   "Allyl..µmol.g.")),
                     scales = "free", nrow = 6, ncol = 3)+
  geom_smooth(method = "lm", alpha = 0.3)+
  ylab("Relative Abunance")
  #ggtitle("ASVs identified by Random Forest as important for predicting\nseed chemistry")

ggsave(rhizoM_cor, filename = paste0(figures_04.fp, "/corr_rhizoM_3.20.png"), height = 9, width = 7)

```


