---
title: "DRAMannot_1"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Load Libraries
```{r}
library(plyr) # always load before tidyverse to avoid conflicts with dplyr
library(tidyverse) # lazily load all of tidyverse, just in case I want to use it.
library(vegan)
library(stringr)

```

## Set up input and output directories
```{r Set up input and output directories, include = FALSE}
project.fp <- "/Users/coin8046/Desktop/FiererLabNotebook/mustard_microbiome/metagenomics/R_analyses"
#raw_data.fp <- file.path(project.fp, "00_rawdata")
tmp_data.fp <- "/Users/coin8046/Desktop/DRAM.mmTESTresults"
clean_data.fp <- file.path(project.fp, "01_clean_data")
    figures_01_clean.fp <- file.path(clean_data.fp, "figures")
    outputs_01_clean.fp <- file.path(clean_data.fp, "outputs")
raw_data.fp <- file.path(project.fp, "00_rawdata_dbs")

if (!dir.exists(clean_data.fp)) {dir.create(clean_data.fp, recursive = TRUE)}    
if (!dir.exists(figures_01_clean.fp)) {dir.create(figures_01_clean.fp, recursive = TRUE)}
if (!dir.exists(outputs_01_clean.fp)) {dir.create(outputs_01_clean.fp, recursive = TRUE)}
```

## read in data
```{r}
mm2_bt2 <- read.table(file = paste0(tmp_data.fp, '/mmT2.bt2.cov.tsv'), sep = '\t', header = TRUE)
mm2_annot <- read.table(file = paste0(tmp_data.fp, '/annotations2.2.tsv'), sep = '\t', header = TRUE, fill = TRUE)

mm1.10 <- read.table(file = paste0(tmp_data.fp, '/mmT1.out/annotationsT10.tsv'), sep = '\t', header = TRUE, fill = TRUE)
mm1.15 <-  read.table(file = paste0(tmp_data.fp, '/mmT1.out/annotationsT15.tsv'), sep = '\t', header = TRUE, fill = TRUE) 
```


## merge_data 
```{r}
mm2_all <- mm2_bt2 %>% 
  inner_join(mm2_annot, by = "rname") %>% 
  select(-c("fasta")) %>% 
  mutate(unr90 = gsub("_.*","", uniref_id)) %>% 
  mutate(unr90.f = paste0("UniRef90_",unr90))

mm2_SN <- mm2_all %>% 
  filter(grepl('sulf|Sulf|nitr|Nitr', uniref_hit))

mm2_glyc <- mm2_all %>% 
  filter(grepl('glucosidase|glycoside hydrolase|myrosinase', uniref_hit))


#mm2_unr <- mm2_all %>% 
  #left_join(unr_db, by = ())

```

### comps
```{r}
mm2.25f <- mm2_annot %>% 
  filter(rank != "E")

mm1.15f <- mm1.15 %>% 
  filter(rank != "E")

mm1.10f <- mm1.10 %>% 
  filter(rank != "E")



#nrow(mm1.15)
```

