---
title: "03_correlations"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Set up

### Load libraries
```{r}
library(mctoolsr)
library(plyr) # always load before tidyverse to avoid conflicts with dplyr
library(tidyverse) # lazily load all of tidyverse, just in case I want to use it.
library(vegan)
library(plotly)
library(readr)
library(stringr)

set.seed(10)
```

### Set up input and output directories
```{r Set up input and output directories, include = FALSE}
project.fp <- "/Users/coin8046/Desktop/FiererLabNotebook/mustard_microbiome/mustard_16S/r_analyses"

clean_data.fp <- file.path(project.fp, "01_clean_data")
    outputs_01_clean.fp <- file.path(clean_data.fp, "outputs")

explr_02.fp <- file.path(project.fp, "02_exploration")
    outputs_02.fp <- file.path(explr_02.fp, "outputs")
    
corr_03.fp <- file.path(project.fp, "03_correlations")
    outputs_03.fp <- file.path(corr_03.fp, "outputs")
    figures_03.fp <- file.path(corr_03.fp, "figures")

if (!dir.exists(corr_03.fp)) {dir.create(corr_03.fp, recursive = TRUE)}    
if (!dir.exists(outputs_03.fp)) {dir.create(outputs_03.fp, recursive = TRUE)}
if (!dir.exists(figures_03.fp)) {dir.create(figures_03.fp, recursive = TRUE)}
```

## Data set up 

### Read in 16S data
```{r}
## clean filt2 data
input_f2 <- readRDS(paste0(outputs_01_clean.fp, "/input_f2.RDS"))

## clean filt2 data relative abundance
input_f2_relab <- readRDS(paste0(outputs_01_clean.fp, "/input_f2_relab.RDS"))

## clean filt2 data rarefied (2000,4000 reads)
input_f2_rar2 <- readRDS(paste0(outputs_01_clean.fp, "/input_f2_rar4k.RDS"))
input_f2_rar1 <- readRDS(paste0(outputs_01_clean.fp, "/input_f2_rar2k.RDS"))

```

### Read in sample metadata
```{r}
metadata.r <- readRDS("/Users/coin8046/Desktop/FiererLabNotebook/mustard_microbiome/mustard_chem/01_outputs/plant.chem_data.RDS")

metadata <- metadata.r[-c(70), ]
sampleIDs <- metadata$Pot.ID

```

### adjust mapping file
```{r}
map.r <- input_f2_rar2$map_loaded %>% 
  mutate(SampleID_3 = gsub('(.*)_\\w+', '\\1', SampleID_2)) %>% 
  mutate(soil_type = as.factor(soil_type))

input_f2_rar2$map_loaded <- map.r
```


##### random diversity checks
```{r}
richness.r <- calc_diversity(tax_table = input_f2_rar2$data_loaded, metric = 'richness') %>% as.data.frame()
names(richness.r) <- "richness"

richness.df.r <- richness.r %>% 
  rownames_to_column(var = "sample_id") %>% 
  mutate(condition = gsub("\\-.*", "", sample_id)) %>% 
  mutate(soil_type = parse_number(condition)) 

richness.df.r[4:23,3] <- "B"

richness.df <- richness.df.r %>% 
  mutate(sterility = gsub("[^a-zA-Z]", "", condition)) %>% 
  mutate(sample_type = gsub(".*_", "", sample_id)) %>% 
  mutate(condition2 = paste(condition, sample_type, sep = "_"))

richness.df[4:23, 4] <- "0"

rich_ord <- c("B_rhizo","B_root","M1_rhizo","S1_rhizo","M1_root","S1_root",
              "M2_rhizo","S2_rhizo","M2_root","S2_root",
              "M3_rhizo","S3_rhizo","M3_root","S3_root",
              "M4_rhizo","S4_rhizo","M4_root","S4_root",
              "M5_rhizo","S5_rhizo","M5_root","S5_root")

ggplot(data = richness.df %>% 
         subset(sample_type %in% c("root", "rhizo")),
       aes(x = condition2, y = richness, 
           shape = sterility, color = as.factor(soil_type)))+
  geom_boxplot()+
  geom_point()+
  scale_x_discrete(limits = rich_ord)+
  theme(axis.text.x = element_text(angle=45, hjust = 1))

```


### Make sample specific dataframes
```{r}

### f2 rarefied data
input_rhizo.r = filter_data(input_f2_rar2, 'sample_type_2', keep_vals = c("rhizo", "rihzo"))
input_rhizo = filter_data(input_rhizo.r, 'SampleID_3', keep_vals = c(sampleIDs))

input_root.r = filter_data(input_f2_rar2, 'sample_type_2', keep_vals = c("root"))
input_root = filter_data(input_root.r, 'SampleID_3', keep_vals = c(sampleIDs))

### select specific taxa of interest




###### add chem to mapping files
map.rz <- input_rhizo$map_loaded %>% 
  left_join(metadata, by = c("SampleID_3" = "Pot.ID"))
input_rhizo$map_loaded <- map.rz

map.rt <- input_root$map_loaded %>% 
  left_join(metadata, by = c("SampleID_3" = "Pot.ID"))
input_root$map_loaded <- map.rt
```

### Summarize taxonomy to create rank dataframes at different levels
```{r}

#class summary
rhizo_class <- summarize_taxonomy(input = input_rhizo, level = 3, report_higher_tax = TRUE)
rhizo_class[is.na(rhizo_class)] <- 0

root_class <- summarize_taxonomy(input = input_root, level = 3, report_higher_tax = TRUE)
root_class[is.na(root_class)] <- 0

#order summary
rhizo_order <- summarize_taxonomy(input = input_rhizo, level = 4, report_higher_tax = TRUE)
rhizo_order[is.na(rhizo_order)] <- 0

root_order <- summarize_taxonomy(input = input_root, level = 4, report_higher_tax = TRUE)
root_order[is.na(root_order)] <- 0

#family summary
rhizo_family <- summarize_taxonomy(input = input_rhizo, level = 5, report_higher_tax = TRUE)
rhizo_family[is.na(rhizo_family)] <- 0

root_family <- summarize_taxonomy(input = input_root, level = 5, report_higher_tax = TRUE)
root_family[is.na(root_family)] <- 0

#genus summary
rhizo_genus <- summarize_taxonomy(input = input_rhizo, level = 6, report_higher_tax = TRUE)
rhizo_genus[is.na(rhizo_genus)] <- 0

root_genus <- summarize_taxonomy(input = input_root, level = 6, report_higher_tax = TRUE)
root_genus[is.na(root_genus)] <- 0

#asv summary
rhizo_asv <- summarize_taxonomy(input = input_rhizo, level = 7, report_higher_tax = TRUE)
rhizo_asv[is.na(rhizo_asv)] <- 0

root_asv <- summarize_taxonomy(input = input_root, level = 7, report_higher_tax = TRUE)
root_asv[is.na(root_asv)] <- 0

```

### chem specific dataframes
```{r}
chem.rz <- map.rz %>% 
  select("sample_id", c(contains(".."), "total_seed_mass")) %>% 
  select(-c("no.flowers..pods.drying", "Updated.Mass..g."))

chem.rt <- map.rt %>% 
  select("sample_id", c(contains(".."), "total_seed_mass")) %>% 
  select(-c("no.flowers..pods.drying", "Updated.Mass..g."))
  
```

### format taxa dfs
```{r}
rt_gen.df <- root_genus %>% 
  t() %>% as.data.frame %>% 
  rownames_to_column(var = "sample_id")

rt_asv.df <- root_asv %>% 
  t() %>% as.data.frame %>% 
  rownames_to_column(var = "sample_id")
  
```

### correlation loop
```{r}
## roots

### this is to get the final table structure in order
#rt.genera <- root_genus %>% rownames()
## this is a list of all the ASVs in my dataframe
rt.asv <- root_asv %>% rownames()

#iterations_rtg = length(rt.genera)
## length of the asv list (so the loop knows how many times to loop)
iterations_rtasv = length(rt.asv)

variables = 2

#rtg_output <- matrix(ncol=variables, nrow=iterations_rtg)
## make an empty matrix to store results
rtasv_output <- matrix(ncol=variables, nrow=iterations_rtasv)

#cor.test(x = rt_gen.df[,2], y = chem.rt[,2], method = "spearman")

for (i in 2:iterations_rtasv){
tmp_cor <- cor.test(x = rt_asv.df[,i], y = chem.rt[,2], method = "spearman")
tmp_row <- c(tmp_cor$p.value, tmp_cor$estimate)
rtasv_output[i,] <- tmp_row 
}
rtasv_output <- as.data.frame(rtasv_output)
names(rtasv_output) <- c("pval", "Rho")
rownames(rtasv_output) <- rt.asv

rtasv_output_fdr <- rtasv_output %>% 
  rownames_to_column(var="asv") %>% 
  mutate(pval.adj = p.adjust(pval, method = "fdr")) %>% 
  mutate(sig = ifelse(pval.adj < 0.001, yes = "***", no = 
                        ifelse(pval.adj < 0.01, yes = "**", no = 
                                 ifelse(pval.adj < 0.05, yes = "*", no = "ns"))))

# reorder columns
rtasv_output_fdr <- rtasv_output_fdr[,c(1,3,2,4,5)]

# reorder rows
rtasv_output_final <- rtasv_output_fdr %>% 
  arrange(pval.adj)
```

