---
title: "03_RandomForest"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Set up

### Load libraries
```{r}
library(mctoolsr)
library(plyr) # always load before tidyverse to avoid conflicts with dplyr
library(tidyverse) # lazily load all of tidyverse, just in case I want to use it.
library(vegan)
library(plotly)
library(readr)
#library(rowr)
library(caret)
library(ranger)
library(stringr)
library(Boruta)

set.seed(10)
```

### Set up input and output directories
```{r Set up input and output directories, include = FALSE}
project.fp <- "/Users/coin8046/Desktop/FiererLabNotebook/mustard_microbiome/mustard_16S/r_analyses"
clean_data.fp <- file.path(project.fp, "01_clean_data")
    outputs_01_clean.fp <- file.path(clean_data.fp, "outputs")

explr_02.fp <- file.path(project.fp, "02_exploration")
    outputs_02.fp <- file.path(explr_02.fp, "outputs")

randfor_04.fp <- file.path(project.fp, "04_RandomForest")
  outputs_04.fp <- file.path(randfor_04.fp, "outputs")
  figures_04.fp <- file.path(randfor_04.fp, "figures")

if (!dir.exists(randfor_04.fp)) {dir.create(randfor_04.fp, recursive = TRUE)}    
if (!dir.exists(outputs_04.fp)) {dir.create(outputs_04.fp, recursive = TRUE)}
if (!dir.exists(figures_04.fp)) {dir.create(figures_04.fp, recursive = TRUE)}
```

## Data set up 

### Read in 16S data
```{r}
# rhizosphere data
## rarefied, filtered data: chem samples only
input_rhizo.c <- readRDS(paste0(outputs_02.fp, "/input_rhizo.chem.RDS"))
input_rhizo.cM <- readRDS(paste0(outputs_02.fp, "/input_rhizo.cM.RDS"))
input_rhizo.cS <- readRDS(paste0(outputs_02.fp, "/input_rhizo.cS.RDS"))

input_root.c <- readRDS(paste0(outputs_02.fp, "/input_root.chem.RDS"))
input_root.cM <- readRDS(paste0(outputs_02.fp, "/input_root.cM.RDS"))
input_root.cS <- readRDS(paste0(outputs_02.fp, "/input_root.cS.RDS"))
```

### Read in sample metadata
```{r}
metadata.r <- readRDS("/Users/coin8046/Desktop/FiererLabNotebook/mustard_microbiome/mustard_chem/01_outputs/plant.chem_data.RDS")

metadata <- metadata.r[-c(70), ]
sampleIDs <- metadata$Pot.ID

```

#### write outputs for python RF modeling

```{r}
rhizoM_ASV <- input_rhizo.cM$data_loaded %>% 
  rownames_to_column(var = "ASV")
rhizoS_ASV <- input_rhizo.cS$data_loaded %>% 
  rownames_to_column(var = "ASV")

rootM_ASV <- input_root.cM$data_loaded %>% 
  rownames_to_column(var = "ASV")
rootS_ASV <- input_root.cS$data_loaded %>% 
  rownames_to_column(var = "ASV")

rhizoM_map <- input_rhizo.cM$map_loaded %>% 
  inner_join(metadata, by = c("SampleID_3" = "Pot.ID"))
rhizoS_map <- input_rhizo.cS$map_loaded %>% 
  inner_join(metadata, by = c("SampleID_3" = "Pot.ID"))

rootM_map <- input_root.cM$map_loaded %>% 
  inner_join(metadata, by = c("SampleID_3" = "Pot.ID"))
rootS_map <- input_root.cS$map_loaded %>% 
  inner_join(metadata, by = c("SampleID_3" = "Pot.ID"))
  
write_csv(x = rhizoM_ASV, file = paste0(outputs_04.fp, "/rhizoM_ASV.csv"))
write_csv(x = rhizoS_ASV, file = paste0(outputs_04.fp, "/rhizoS_ASV.csv"))
write_csv(x = rootM_ASV, file = paste0(outputs_04.fp, "/rootM_ASV.csv"))
write_csv(x = rootS_ASV, file = paste0(outputs_04.fp, "/rootS_ASV.csv"))

write_csv(x = rhizoM_map, file = paste0(outputs_04.fp, "/rhizoM_map.csv"))
write_csv(x = rhizoS_map, file = paste0(outputs_04.fp, "/rhizoS_map.csv"))
write_csv(x = rootM_map, file = paste0(outputs_04.fp, "/rootM_map.csv"))
write_csv(x = rootS_map, file = paste0(outputs_04.fp, "/rootS_map.csv"))


```


### select most abundant ASVs (optional)
```{r}
## calculate top rhizso ASVs
rhizo.df <- input_rhizo.c$data_loaded %>% 
  mutate(asv_reads = rowSums(.[1:ncol(input_rhizo.c$data_loaded)])) %>% 
  mutate(asv_ubiq = rowSums(.[1:ncol(input_rhizo.c$data_loaded-1)] !=0)) %>% 
  rownames_to_column(var = "ASV")

rzASVs_45 <- rhizo.df %>% 
  filter(asv_ubiq>45) %>% 
  select("ASV") %>% 
  unlist()


## calculate top root ASVs
root.df <- input_root.c$data_loaded %>% 
  mutate(asv_reads = rowSums(.[1:ncol(input_root.c$data_loaded)])) %>% 
  mutate(asv_ubiq = rowSums(.[1:ncol(input_root.c$data_loaded-1)] !=0)) %>% 
  rownames_to_column(var = "ASV")

rtASVs_35 <- rhizo.df %>% 
  filter(asv_ubiq>35) %>% 
  select("ASV") %>% 
  unlist()

```

```{r}
input_rhizo.filt <- filter_taxa_from_input(input = input_rhizo.c, taxa_IDs_to_keep = rzASVs_45)

input_rhizo.fc.M = filter_data(input_rhizo.filt, 'sterility', keep_vals = c("M"))
input_rhizo.fc.S = filter_data(input_rhizo.filt, 'sterility', keep_vals = c("S"))

### or should I make M and S dfs separately?
input_rhizo.fc2.M = filter_taxa_from_input(input_rhizo.cM, filter_thresh = 5)
input_rhizo.fc2.S = filter_taxa_from_input(input_rhizo.cS, filter_thresh = 5)

input_root.filt <- filter_taxa_from_input(input = input_root.c, taxa_IDs_to_keep = rtASVs_35)

input_root.fc.M = filter_data(input_root.filt, 'sterility', keep_vals = c("M"))
input_root.fc.S = filter_data(input_root.filt, 'sterility', keep_vals = c("S"))

### or should I make M and S dfs separately?
input_root.fc2.M = filter_taxa_from_input(input_root.cM, filter_thresh = 4)
input_root.fc2.S = filter_taxa_from_input(input_root.cS, filter_thresh = 4)
```

NOTE: summarize taxonomy for any of these?

## Prep dataframes for RF
```{r}
#input_rhizo.cM

ASV_rhizo.M <- summarize_taxonomy(input_rhizo.cM, level = 7, report_higher_tax =  FALSE) %>% 
  t() %>% as.data.frame() %>% 
  rownames_to_column(var = "sample_id") %>% 
  inner_join(input_rhizo.cM$map_loaded, by = c("sample_id"= "SampleID_2"))
  #inner_join(metadata, by = c("SampleID_3" = "Pot.ID"))

# keep "Allyl..µmol.g."
# "ASV_1036" "ASV_1205" "ASV_128"  "ASV_1825" "ASV_190"  "ASV_22"   "ASV_2217" "ASV_302"  "ASV_316"  "ASV_369" "ASV_53"   "ASV_584"  "ASV_687"  "ASV_808"  "ASV_104"  "ASV_11"   "ASV_278"  "ASV_400"  "ASV_625"

ASV_rhizo.M.rf <- ASV_rhizo.M %>% 
  column_to_rownames(var = "sample_id") %>% 
  # dplyr::select(c("ASV_128", "ASV_190", "ASV_316", "ASV_808",
  #        "ASV_104", "ASV_11", "ASV_278", "ASV_400", "ASV_625", "ASV_1036",
  #        "ASV_22", "ASV_1205","Allyl..µmol.g.", "sample_id")) %>%
  #          column_to_rownames(var = "sample_id")
  #  select(c("ASV_112", "ASV_228", "ASV_278", "ASV_299", "ASV_302", "ASV_316", "ASV_336", "ASV_369",
  #          "ASV_44",  "ASV_48",  "ASV_53",  "ASV_607", "ASV_692", "ASV_697", "ASV_745", "ASV_926",
  #        "Allyl..µmol.g.", "sample_id")) %>%
  dplyr::select(-c("Plate","Well","sample_type_1","sample_type_2", "ReadCount.raw","ReadCount.filt",
            "condition","soil_type.x","sterility.x","SampleID_3","flower.buds.appear","full.flower.appear",
            "seed.pods.appear","no.flowers..pods.drying","Spider.mites","total.seed.count","Last.Seeds.Collected",
            "Root.Rhizo.Harvest","total_seed_mass","sample.name","height_9","height_15","Updated.Mass..g.",
            "X3.Butenyl..µmol.g.","X3MT..µmol.g.","Indoles..µmol.g.","treatment","soil_type.y","sterility.y"))
ASV_rhizo.M.rf


### write df for more python play
ASV_rhizo.M.eli <- ASV_rhizo.M.rf %>%
  dplyr::select(Allyl..µmol.g., everything())

write_csv(x = ASV_rhizo.M.eli, file = paste0(outputs_04.fp, "/ASV_rhizo.M.eli.csv"))
```

#### S dfs
```{r}

ASV_rhizo.S <- summarize_taxonomy(input_rhizo.cS, level = 7, report_higher_tax =  FALSE) %>% 
  t() %>% as.data.frame() %>% 
  rownames_to_column(var = "sample_id") %>% 
  inner_join(input_rhizo.cS$map_loaded, by = "sample_id") %>% 
  inner_join(metadata, by = c("SampleID_3" = "Pot.ID"))

# "ASV_745" not present in S dataset

ASV_rhizo.S.rf <- ASV_rhizo.S %>% 
  select(c("ASV_107",  "ASV_1255", "ASV_33" ,  "ASV_34",   "ASV_3804", 
           "ASV_465",  "ASV_534",  "ASV_557", "ASV_68",
           "Allyl..µmol.g.", "sample_id")) %>% 
  column_to_rownames(var = "sample_id")



ASV_rhizo.S.rf1 <- ASV_rhizo.S %>% 
dplyr::select(-c("Plate","Well","sample_type_1","sample_type_2","SampleID_2","ReadCount.raw","ReadCount.filt",
          "condition","soil_type.x","sterility.x","SampleID_3","flower.buds.appear","full.flower.appear",
          "seed.pods.appear","no.flowers..pods.drying","Spider.mites","total.seed.count","Last.Seeds.Collected",
          "Root.Rhizo.Harvest","total_seed_mass","sample.name","height_9","height_15","Updated.Mass..g.",
          "X3.Butenyl..µmol.g.","X3MT..µmol.g.","Indoles..µmol.g.","treatment","soil_type.y","sterility.y")) %>%
  column_to_rownames(var = "sample_id")

```


### Root dataframes for rf
```{r}

ASV_root.M <- summarize_taxonomy(input_root.cM, level = 7, report_higher_tax =  FALSE) %>% 
  t() %>% as.data.frame() %>% 
  rownames_to_column(var = "sample_id") %>% 
  inner_join(input_root.cM$map_loaded, by = "sample_id") %>% 
  inner_join(metadata, by = c("SampleID_3" = "Pot.ID"))

ASV_root.M.rf <- ASV_root.M %>% 
  ### rhizo boruta ASVs
  # select(c("ASV_1036", "ASV_1205", "ASV_128",  "ASV_1825", "ASV_190",  "ASV_22",
  #          "ASV_2217", "ASV_302",  "ASV_316",  "ASV_369", "ASV_53",   "ASV_584",
  #          "ASV_687",  "ASV_808",  "ASV_104",  "ASV_11",   "ASV_278",  "ASV_400",  "ASV_625",
  #          "Allyl..µmol.g.", "sample_id")) %>%
  ### root boruta.tr ASVs
  # select(c("ASV_11", "ASV_14", "ASV_157", "ASV_187","ASV_190", "ASV_38" ,"ASV_422",
  #          "ASV_48",  "ASV_49",  "ASV_88",  "ASV_96", "Allyl..µmol.g.", "sample_id")) %>% 
  ### root boruta ASVs (all data)
  # select(c("ASV_130",  "ASV_14",  "ASV_143","ASV_145",  "ASV_1478", "ASV_325",
  #          "ASV_38",   "ASV_48", "ASV_88","Allyl..µmol.g.", "sample_id")) %>% 
  ### root boruta ASVs (combined)
  # select(c(all_of(all_asvs.rt), "Allyl..µmol.g.", "sample_id")) %>%
  ### allll root ASVs
  dplyr::select(-c("Plate","Well","sample_type_1","sample_type_2","SampleID_2","ReadCount.raw","ReadCount.filt",
            "condition","soil_type.x","sterility.x","SampleID_3","flower.buds.appear","full.flower.appear",
            "seed.pods.appear","no.flowers..pods.drying","Spider.mites","total.seed.count","Last.Seeds.Collected",
            "Root.Rhizo.Harvest","total_seed_mass","sample.name","height_9","height_15","Updated.Mass..g.",
            "X3.Butenyl..µmol.g.","X3MT..µmol.g.","Indoles..µmol.g.","treatment","soil_type.y","sterility.y")) %>%
  column_to_rownames(var = "sample_id")

## input_root.cS

ASV_root.S <- summarize_taxonomy(input_root.cS, level = 7, report_higher_tax =  FALSE) %>% 
  t() %>% as.data.frame() %>% 
  rownames_to_column(var = "sample_id") %>% 
  inner_join(input_root.cS$map_loaded, by = "sample_id") %>% 
  inner_join(metadata, by = c("SampleID_3" = "Pot.ID"))

ASV_root.S.rf <- ASV_root.S %>% 
  ## allll root ASVs
  select(-c("Plate","Well","sample_type_1","sample_type_2","SampleID_2","ReadCount.raw","ReadCount.filt",
            "condition","soil_type.x","sterility.x","SampleID_3","flower.buds.appear","full.flower.appear",
            "seed.pods.appear","no.flowers..pods.drying","Spider.mites","total.seed.count","Last.Seeds.Collected",
            "Root.Rhizo.Harvest","total_seed_mass","sample.name","height_9","height_15","Updated.Mass..g.",
            "X3.Butenyl..µmol.g.","X3MT..µmol.g.","Indoles..µmol.g.","treatment","soil_type.y","sterility.y")) %>%
  ### Mroot boruta ASVs (combined)
  #select(c(all_of(all_asvs.rt), "Allyl..µmol.g.", "sample_id")) %>%
  ### Sroot ASVs (misc)
   # select(c("ASV_208", "ASV_26", "ASV_418", "ASV_549", "ASV_787", "ASV_164", "ASV_195", "ASV_201", "ASV_24", 
   #          "ASV_386", "ASV_389", "ASV_4", "ASV_484", "ASV_664", "ASV_807", "ASV_88", 
   #          "Allyl..µmol.g.", "sample_id")) %>% 
  ### Sroot ASVs (misc2)
  # select(c("ASV_2294", "ASV_2429", "ASV_549","ASV_1725", "ASV_2481", "ASV_25", "ASV_549","ASV_208", 
  #          "ASV_2414", "ASV_25", "ASV_318", "ASV_801","Allyl..µmol.g.", "sample_id")) %>% 
  column_to_rownames(var = "sample_id")
```


## Boruta feature selection
```{r}

#https://academic.oup.com/bib/article/20/2/492/4554516
#https://www.machinelearningplus.com/machine-learning/feature-selection/

# Perform Boruta search
boruta_output.rz.M <- Boruta(Allyl..µmol.g. ~., 
                        data=na.omit(ASV_rhizo.M.rf), doTrace=0) 

boruta_output.rz.M
boruta_signif.rz.M <- getSelectedAttributes(boruta_output.rz.M, withTentative = TRUE)
### train rhizoM data1
  #"ASV_112" "ASV_228" "ASV_278" "ASV_299" "ASV_302" "ASV_316" "ASV_336" "ASV_369" 
  #"ASV_44"  "ASV_48"  "ASV_53"  "ASV_607" "ASV_692" "ASV_697" "ASV_745" "ASV_926"
### train rhizoM data2
  # "ASV_112"  "ASV_1205" "ASV_128"  "ASV_2377" "ASV_278"  "ASV_3098" "ASV_336" 
  #"ASV_369"  "ASV_44"   "ASV_48"   "ASV_53" "ASV_5718" "ASV_607"  "ASV_692"  "ASV_745"  "ASV_926"
### all rhizoM data1
  # "ASV_1036" "ASV_11"   "ASV_1205" "ASV_128"  "ASV_190"  "ASV_22"   "ASV_2217" "ASV_2377" "ASV_369"  "ASV_400"  "ASV_449" 
  # "ASV_46"   "ASV_584"  "ASV_745" 
### all rhizoM data2
  # "ASV_1036" "ASV_1205" "ASV_128"  "ASV_1825" "ASV_190"  "ASV_2139" "ASV_22"   "ASV_2217" "ASV_2377" "ASV_278"  "ASV_369" 
  # "ASV_449"  "ASV_46"   "ASV_53"   "ASV_584"  "ASV_745"  "ASV_76"   "ASV_8"

boruta_output.rt.M <- Boruta(Allyl..µmol.g. ~., 
                        data=na.omit(ASV_root.M.rf), doTrace=0) 
boruta_output.rt.M
boruta_signif.rt.M <- getSelectedAttributes(boruta_output.rt.M, withTentative = TRUE)
### all rootM data1
  # "ASV_130" "ASV_14"  "ASV_143" "ASV_166" "ASV_190" "ASV_22"  "ASV_229" "ASV_314" "ASV_367" "ASV_48"  "ASV_88"
### all rootM data2
  # "ASV_130"  "ASV_14"   "ASV_1852" "ASV_238"  "ASV_4"    "ASV_422"  "ASV_48"   "ASV_88"   "ASV_943"  "ASV_96"



### all rootM data burata output
 # 5 attributes confirmed important: ASV_130, ASV_14, ASV_143, ASV_48, ASV_88;
 # 4 tentative attributes left: ASV_145, ASV_1478, ASV_325, ASV_38;
### all rootS data burata output
 # 3 attributes confirmed important: ASV_2294, ASV_2429, ASV_549;
 # 4 attributes confirmed important: ASV_1725, ASV_2481, ASV_25, ASV_549;
 # 4 tentative attributes left: ASV_208, ASV_2414, ASV_25, ASV_318;
 # 1 tentative attributes left: ASV_801;


boruta_output.trS <- Boruta(Allyl..µmol.g. ~., 
                        data=na.omit(train), doTrace=0) 

boruta_output.trS
boruta_signif.trS <- getSelectedAttributes(boruta_output.trS, withTentative = TRUE)
### train rootM data burata output
# 2 attributes confirmed important: ASV_157, ASV_38;
# 9 tentative attributes left: "ASV_11" "ASV_14""ASV_187" "ASV_190" "ASV_422" "ASV_48"  "ASV_49"  "ASV_88"  "ASV_96"
### train rootS data burata output
# 1 attributes confirmed important: ASV_302;
# 5 attributes confirmed important: ASV_208, ASV_26, ASV_418, ASV_549, ASV_787;
# 11 tentative attributes left: ASV_164, ASV_195, ASV_201, ASV_24, ASV_386,
#                               ASV_389, ASV_4, ASV_484, ASV_664, ASV_807, ASV_88
#4 tentative attributes left: ASV_113, ASV_163, ASV_2429, ASV_311;



intersect(boruta_signif, boruta_signif.tr)
all_asvs.rt <- union(boruta_signif, boruta_signif.tr)

intersect(boruta_signif.trS, boruta_signif.rt.S)
all_asvs.rtS <- union(boruta_signif.trS, boruta_signif.rt.S)

intersect(all_asvs.rt, all_asvs.rtS)


### other misc ways of looking at things
roughFixMod <- TentativeRoughFix(boruta_output)
boruta_signif <- getSelectedAttributes(roughFixMod)

# Variable Importance Scores
imps <- attStats(boruta_output)
#imps2 = imps[imps$decision != 'Rejected', c('meanImp', 'decision')]
imps[order(-imps$meanImp), ]  # descending sort

```


## Random forest training

```{r}
### Separate training and testing data
set.seed(13)
random_ordered <- ASV_rhizo.M.rf[sample(nrow(ASV_rhizo.M.rf)),]
number_training_samples <- ceiling(nrow(random_ordered) * 0.6)
train <- random_ordered[1:number_training_samples,]
test <- random_ordered[(number_training_samples + 1):nrow(random_ordered),]

random_orderedS <- ASV_rhizo.S.rf[sample(nrow(ASV_rhizo.S.rf)),]
number_training_samplesS <- ceiling(nrow(random_orderedS) * 0.6)
trainS <- random_orderedS[1:number_training_samplesS,]
testS <- random_orderedS[(number_training_samplesS + 1):nrow(random_orderedS),]

### Create random forest models
ranger1rzM1 <- ranger(Allyl..µmol.g. ~.,
                  data=train,
                  num.trees = 1000,
                  splitrule = "variance",
                  write.forest = TRUE, 
                  #min.node.size = 5, 
                  classification = FALSE, 
                  seed = 10)
print(ranger1rzM1)



### another way to create RF model
control <- trainControl(method='repeatedcv', 
                        number=10, 
                        repeats=3)
mtry <- sqrt(ncol(train))
tunegrid <- expand.grid(.mtry=mtry)

train1 <- train(Allyl..µmol.g. ~.,
                data=train,
                method="rf")
                #tuneGrid=tunegrid)
                #trControl=control)
print(train1)
varImp(train1)




```

## test predictions
```{r}
pred <- predict(ranger1rzM1, data = test)
cor.test(pred$predictions, test$Allyl..µmol.g.)

pred2 <- predict(train1, newdata = test) %>% as.data.frame() %>% rownames_to_column(var = "sample_num")
cor.test(pred2$., test$Allyl..µmol.g.)

# predS <- predict(ranger1rzM1, data = ASV_rhizo.S.rf1[,1:3781])
# cor.test(predS$predictions, ASV_rhizo.S.rf1$Allyl..µmol.g.)
# 
# resS <- as.data.frame(ASV_rhizo.S.rf1$Allyl..µmol.g.)%>% 
#   rownames_to_column(var = "test_num")
# 
# checkS <- predS$predictions %>% 
#   as.data.frame() %>% 
#   rownames_to_column(var = "test_num") %>%
#   inner_join(resS, by = "test_num")


pred.1 <- pred$predictions %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "test_num")

check1 <- test %>% 
  rownames_to_column(var = "sample_num") %>% 
  rownames_to_column(var = "test_num") %>% 
  select(c("Allyl..µmol.g.", "sample_num", "test_num")) %>% 
  #inner_join(pred, by= "test_num") %>% 
  inner_join(pred2, by = "sample_num")

#cor.test(check$., check$Allyl..µmol.g.)
cor.test(check1$., check1$Allyl..µmol.g.)
#cor.test(check$..y, check$Allyl..µmol.g.)

predcor <- ggplot(check1, aes(x=., y=Allyl..µmol.g.))+ #Allyl..µmol.g.
  geom_point()+
  geom_smooth(method = "lm")+
  #geom_abline(slope = 1, intercept = 0, color = 'red')+
  ylab("true proportion of Allyl in seeds")+
  xlab("predicted proportion Allyl in seeds")+
  #labs(title = "Predicted by slurry bacterial family composition")+
  theme_bw(base_size = 18)+
  ggtitle("Predictive accuracy of Random Forest model")

predcor

ggsave(predcor, filename = paste0(figures_04.fp, "/predcor.png"))
```

## plot correlations
```{r}

### some top important ASVs as output from RF models
#ASVs 190, 808, 247, 80, 1036, 400, 437, 11, 46, 449
#ASVs 64, 192, 278, 115, 427
#ASVs 808, 190, 1036, 316, 437, 80, 104, 128, 201, 223

cor.test(ASV_rhizo.M.rf$Allyl..µmol.g., ASV_rhizo.M.rf$ASV_190, method = "pearson")
cor.test(ASV_rhizo.M.rf$Allyl..µmol.g., ASV_rhizo.M.rf$ASV_11, method = "pearson")


ASV_rhizo.M.rf_long <- ASV_rhizo.M.rf %>% 
  pivot_longer(! Allyl..µmol.g., 
               names_to = "ASV", values_to = "abun")

ASV_root.rf_long <- ASV_root.M.rf %>% 
  pivot_longer(! Allyl..µmol.g., 
               names_to = "ASV", values_to = "abun")


rhizoM_cor <- ggplot(data = ASV_rhizo.M.rf_long %>% 
         filter(ASV %in% c("ASV_128", "ASV_190", "ASV_316", "ASV_808",
         "ASV_104", "ASV_11", "ASV_278", "ASV_400", "ASV_625", "ASV_1036",
         "ASV_22", "ASV_1205")), 
                           #"ASV_589", "ASV_79", "ASV_61", "ASV_6")), 
       mapping = aes(x = Allyl..µmol.g., y = abun))+
  geom_point()+
  facet_wrap(~ ASV, scales = "free")+
  geom_smooth(method = "lm", alpha = 0.3)+
  ylab("Relative Abunance")+
  ggtitle("ASVs identified by machine learning methods important for predicting\nseed chemistry")

ggsave(rhizoM_cor, filename = paste0(figures_04.fp, "/corr_rhizoM.png"))


 ggplot(data = ASV_root.rf_long %>% 
         filter(ASV %in% c("ASV_190", "ASV_11", "ASV_22")),
       mapping = aes(x = Allyl..µmol.g., y = abun))+
  geom_point()+
  facet_wrap(~ ASV, scales = "free")+
  geom_smooth(method = "lm", alpha = 0.3)



```
### Roow v Rhizosphere
```{r}
### anything in common between root and rhizo ASVs?
rhizo_asvs <- unique(c("ASV_1036", "ASV_11",   "ASV_1205", "ASV_128",  "ASV_190",
                     "ASV_22",   "ASV_2217", "ASV_2377", "ASV_369",  "ASV_400",
                     "ASV_449", "ASV_46",   "ASV_584",  "ASV_745", "ASV_1036",
                     "ASV_1205", "ASV_128",  "ASV_1825", "ASV_190",  "ASV_2139",
                     "ASV_22",   "ASV_2217", "ASV_2377", "ASV_278",  "ASV_369",
                     "ASV_449",  "ASV_46",   "ASV_53" ,  "ASV_584",  "ASV_745",
                     "ASV_76",   "ASV_8"))
root_asvs <- unique(c("ASV_130", "ASV_14",  "ASV_143", "ASV_166", "ASV_190",
                    "ASV_22",  "ASV_229", "ASV_314", "ASV_367", "ASV_48",
                    "ASV_88", "ASV_130",  "ASV_14", "ASV_1852", "ASV_238",
                    "ASV_4",  "ASV_422",  "ASV_48", "ASV_88", "ASV_943","ASV_96"))
  
intersect(rhizo_asvs, root_asvs)



### anything in common between S and M important ASVs?
S_ASVs <- c("ASV_107",  "ASV_1255", "ASV_33" ,  "ASV_34",   "ASV_3804","ASV_465",  "ASV_534",  "ASV_557", "ASV_68") 

M_ASVs <- c("ASV_1036", "ASV_1205","ASV_128","ASV_1825","ASV_190","ASV_22","ASV_2217","ASV_302","ASV_316","ASV_369","ASV_53","ASV_584","ASV_687")

M2_ASVs <- c("ASV_128", "ASV_190", "ASV_316", "ASV_808", "ASV_104", "ASV_11", "ASV_278", "ASV_400", "ASV_625")

intersect(M_ASVs, M2_ASVs)
union(M_ASVs, M2_ASVs)

intersect(S_ASVs, M_ASVs)
```

### taxonomic investigation
```{r}
M_ASVs.f <- c("ASV_128", "ASV_190", "ASV_316", "ASV_808",
         "ASV_104", "ASV_11", "ASV_278", "ASV_400", "ASV_625", "ASV_1036",
         "ASV_22", "ASV_1205")

S_ASVs.f <- c("ASV_107",  "ASV_1255", "ASV_33" ,  "ASV_34",   "ASV_3804","ASV_465",  "ASV_534",  "ASV_557", "ASV_68")

b_asvs <- c("ASV_11", "ASV_22", "ASV_190")

gbr_asvs <- c("ASV_1036", "ASV_190", "ASV_584", "ASV_1205", "ASV_29")

input_M.ASVs <- filter_taxa_from_input(input = input_rhizo.cM, taxa_IDs_to_keep = M_ASVs.f)
View(input_M.ASVs$taxonomy_loaded)

input_S.ASVs <- filter_taxa_from_input(input = input_rhizo.cS, taxa_IDs_to_keep = S_ASVs.f)
View(input_S.ASVs$taxonomy_loaded)

input_gbr.ASVs <- filter_taxa_from_input(input = input_rhizo.cM, taxa_IDs_to_keep = gbr_asvs)
View(input_gbr.ASVs$taxonomy_loaded)
```


Notes:
repeat with S data (filtered and non-filtered)
other attempts at feature selection? / expand? 
  could also use correlation table to inform ASVs to include (top XX ASVs before correction?)
other levels of taxonomy? (family, genus, etc)
repeat for roots
predict with "good" rf models
   limited test ability due to size of dataset...
   
   
   
   
```{r}

all_data <- readRDS(paste0(outputs_01_clean.fp, "/input_f2.RDS"))

seeds <- filter_data(all_data, filter_cat = "sample_type_2", keep_vals = "seed")
                    
seeds$data_loaded %>% colSums() %>% sort()

summary(seeds$data_loaded %>% colSums())

# library(mctoolsr)
# library(tidyverse)
# 
# pss_data <- readRDS("/Users/coin8046/Desktop/pss_f5_full.rds")
# 
# slurry_data <- filter_data(pss_data, filter_cat = 'sample_type_1', keep_vals = 'microbial')
# slurry_asv <- slurry_data$data_loaded
# 
# slurry_map <- slurry_data$map_loaded %>% 
#   select(-c("sample_type_1", "sample_type_2", "process_type",  "rep_num","rep_cat","ReadCount_nt",  "richness_nt","richness_10","richness_5"))
# 
# slurry_data$map_loaded <- slurry_map
# 
# saveRDS(object = slurry_data, file = "/Users/coin8046/Desktop/slurries_CW.rds")
# 
# saveRDS(slurry_map, file = "/Users/coin8046/Desktop/slurryMetadata.rds")
# saveRDS(slurry_asv, file = "/Users/coin8046/Desktop/slurryASVtab.rds")
# saveRDS(slurry_data$taxonomy_loaded, file = "/Users/coin8046/Desktop/slurrytaxonomy.rds")


```
   

